### PR TITLE
Refactor RunDialog to depend less on RunModel

### LINKED ISSUE #3024: Make GUI independent of RunModel
After the evaluator was introduced, the GUI now gets most of the information about the currently running experiment from the events that it receives (from the EvaluatorTracker). However, there is still some information that is accessed directly from the run model in the RunDialog ([run_dialog.py](https://github.com/equinor/ert/blob/main/ert_gui/simulation/run_dialog.py)). This is mostly related to getting the mask of active realizations and tiem onformation about the entire experiment. We should not depend on getting this information directly from the run model, and instead also pass the necessary information via events.

### PR DESCRIPTION
**Pull request by xjules**
**Issue**
Resolves #3024 

**Approach**
- [x] Refactor `RunDialog` by migrating simulation running info into `BaseRunModel`
- [x] Move more processing logic into RunModel
~~- [ ] Make `EvaluatorTracker` as a proxy for `BaseRunModel`~~ this might be affected by #2132 so left it for now

## Pre review checklist

- [x] Added appropriate labels

Adding labels helps the maintainers when writing release notes, see sections and the
corresponding labels here: https://github.com/equinor/ert/blob/main/.github/release.yml

### CODE REVIEW

**Review by jondequinor — COMMENTED** [review]
Did you figure out the segfault?

**Comment by xjules** [comment]
> Did you figure out the segfault?

Yes, I did. Rebasing commits one by one did the trick 😄

File: ert_shared/models/base_run_model.py [hunk 1]
```diff
@@ -74,21 +83,59 @@ def reset(self) -> None:
...
+
+    def has_failed_realizations(self) -> bool:
+        completed = self.completed_realizations_mask
+        initial = self.initial_realizations_mask
+        for (index, successful) in enumerate(completed):
+            if initial[index] and not successful:
+                return True
+        return False
+
+    def create_mask_from_failed_realizations(self) -> BoolVector:
```
Comment (jondequinor) [inline ert_shared/models/base_run_model.py:104]:
```suggestion
    def _create_mask_from_failed_realizations(self) -> BoolVector:
```

File: ert_shared/models/base_run_model.py [hunk 2]
```diff
@@ -74,21 +83,59 @@ def reset(self) -> None:
...
+        :return: Type BoolVector
+        """
+        completed = self.completed_realizations_mask
+        initial = self.initial_realizations_mask
+        inverted_mask = BoolVector(default_value=False)
+        for (index, successful) in enumerate(completed):
+            inverted_mask[index] = initial[index] and not successful
+        return inverted_mask
+
+    def count_successful_realizations(self) -> int:
```
Comment (jondequinor) [inline ert_shared/models/base_run_model.py:116]:
```suggestion
    def _count_successful_realizations(self) -> int:
```

**Review by jondequinor — APPROVED** [review]
This is solid stuff. Really good refactor. Had some minor comments to rein in the BRM API.

Tested out all run models in the GUI, seems to work well! Also tried restart on a failed mda run.

👏 👏

**Comment by sondreso** [comment]
Could you shorten the commit message to be < 50 characters :slightly_smiling_face:

**Pull request merged**